src/data_gradients/datasets/FolderProcessor.py

```diff
@@ -117,3 +116,107 @@ def __getitem__(self, index):
     def __iter__(self):
         for image_label_file in self.images_with_labels_files:
             yield image_label_file
+
+
+class ImageLabelConfigIterator:
+    """Iterate over all image and label files in the provided directories."""
+
+    def __init__(
+        self,
+        images_dir: str,
+        labels_dir: str,
+        config_path: str,
+        label_extensions: Sequence[str],
+        image_extensions: Sequence[str] = DEFAULT_IMG_EXTENSIONS,
+        verbose: bool = True,
+    ):
+        """
+        :param images_dir:      The directory containing the images.
+        :param labels_dir:      The directory containing the labels.
+        :param label_extensions: The extensions of the labels. Only the labels with this extensions will be considered.
+        :param image_extensions: The extensions of the images. Only the images with this extensions will be considered.
+        :param verbose:         Whether to print extra messages.
+        """
+        self.images_with_labels_files = self.get_image_and_label_file_names(
+            images_dir=images_dir,
+            labels_dir=labels_dir,
+            config_path=config_path,
+            label_extensions=label_extensions,
+            image_extensions=image_extensions,
+            verbose=verbose,
+        )
+
+    def get_image_and_label_file_names(
+        self,
+        images_dir: str,
+        labels_dir: str,
+        config_path: str,
+        image_extensions: List[str],
```

Comment:
Is this typing correct ?

Reply:
I changed it to Sequence to be more general